src/Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore/Internal/DiagnosticsEntityFrameworkCoreLoggerExtensions.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.Internal
+{
+    internal static class DiagnosticsEntityFrameworkCoreLoggerExtensions
+    {
+        private static readonly Action<ILogger, Exception> _noContextType;
+        private static readonly Action<ILogger, Type, Exception> _invalidContextType;
+        private static readonly Action<ILogger, string, Exception> _contextNotRegistered;
+        private static readonly Action<ILogger, string, Exception> _requestPathMatched;
+        private static readonly Action<ILogger, string, Exception> _applyingMigrations;
+        private static readonly Action<ILogger, string, Exception> _migrationsApplied;
+        private static readonly Action<ILogger, string, Exception, Exception> _migrationsEndPointMiddlewareException;
+        private static readonly Action<ILogger, Type, Exception> _attemptingToMatchException;
+        private static readonly Action<ILogger, string, Exception> _contextNotRegisteredDatabaseErrorPageMiddleware;
+        private static readonly Action<ILogger, Exception> _notRelationalDatabase;
+        private static readonly Action<ILogger, Exception> _noRecordedException;
+        private static readonly Action<ILogger, Exception> _noMatch;
+        private static readonly Action<ILogger, Exception> _matched;
+        private static readonly Action<ILogger, Exception, Exception> _databaseErrorPageMiddlewareException;
+
+        static DiagnosticsEntityFrameworkCoreLoggerExtensions()
+        {
+            _noContextType = LoggerMessage.Define(
```

Comment:
I've been putting a comment at each section of log messages with the logger category name to make it obvious why the event id's reset etc.